### PR TITLE
#17494 review slack package

### PR DESCRIPTION
**Pull request by germanm98**
Pull-request for issue #17505 created by @germanm98

## What it does?

Review the Slack package, added need listeners

## How to test it?
Create a slack app at https://api.slack.com/apps
Add the package to a existing app and set it up.
In your Slack app, subscribe your bot to some slack **events**,
set and **interactive message** and also a **slash command**.
Check that all the webhooks mentioned above arrives to your app.

## Technical notes

There are no technical notes.

## Deployment notes

There are no deployment notes.

### CODE REVIEW

File: listeners/webhooks.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        }
+    },
+    callback: function(event) {
+        sys.logs.info('[slack] Received Slack webhook. Processing and triggering a package event.');
+        if (pkg.slack.utils.verifyToken(event.data.body.token)) {
+            if (body.type === 'url_verification') {
+                sys.logs.info('[slack] Valid url verification. Sending challenge back.');
+                return JSON.stringify({challenge: event.data.body.challenge});
+            } else {
+                sys.logs.info('Valid slack event received. Triggering event.');
```
Comment (pasaperez-work) [inline listeners/webhooks.js]:
```suggestion
                sys.logs.info('[slack] Valid slack event received. Triggering event.');
```

File: package.json [hunk 2]
```diff
@@ -26,20 +26,20 @@
...
+            "description": "Token provided to verify that requests are actually coming from Slack (Basic Information -> Verification Token)",
             "type": "text"
         },
         {
             "name": "eventsUrl",
             "label": "Events URL",
             "description": "This is the URL you need to set as 'Request URL' in 'Event Subscriptions' page of your app. Keep in mind that the endpoint must be deployed in order to interact with the Slack configuration page.",
             "type": "label",
-            "value": "config.WEBHOOK_URL+'/events'"
+            "value": "config.WEBHOOK_SYNC_URL+'/events'"
```
Comment (pasaperez-work) [inline package.json]:
```suggestion
            "value": "config.WEBHOOK_SYNC_URL+'/events/'"
```

**Review by jsanchez-slingr — CHANGES_REQUESTED** [review]
Remove unnused http methods from user.js and bot.js
Remove the initial table from README.md.
Keep the same format we have for other packages like "zoho" for the package settings
I think we could add some examples for the bot api, slash commands and interactive messages.
Review titles rules for documentation Check resolved comments on this PR: https://github.com/slingr-stack/google-storage-package/pull/1#discussion_r1790138744
And review the Capital case format for titles.
Remove "(latest version)" for HTTP Service dependency
The slack API link does redirect to a 404 page.
Use chatgpt for improving grammar and spelling

**Comment by pasaperez-work** [comment]
image
Just a small request, could you add the link where each webhook should be configured?
Or a route something like Applications -> webhook-> menu1 -> option 23

**Review by angel-perez-itti — APPROVED** [review]
Great! Thanks @germanm98

**Pull request merged**